tsdb/db.go

```diff
@@ -621,6 +621,18 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 	var wlog *wal.WAL
 	segmentSize := wal.DefaultSegmentSize
+
+	defer func() {
+		// Close files if startup fails somewhere
+		if err == nil {
+			return
+		}
+
+		if er := db.Close(); er != nil {
```

Comment:
```suggestion
		if err := db.Close(); err != nil {
```
if you don't want to shadow `err` you can od `dbErr`  - but not `er`

Reply:
I named `er` based on what I saw in a [piece of code](https://github.com/prometheus/prometheus/blob/416e3e0c3f620d4f6129305e50fe935c03f44b9e/tsdb/db.go#L1511) while working on another PR, but I agree with you.
I'll fix it on both PRs.